src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx

```diff
@@ -1,227 +1,40 @@
-
-  return (
-    <>
-      <ContentHeader
-        title={t.addNewAccessCode}
-        subheading={device.properties.name}
+      <AccessCodeAddForm
+        deviceId={deviceId}
         onBack={onBack}
+        className={className}
       />
-      <div className='seam-main'>
-        <FormField>
-          <InputLabel>{t.nameInputLabel}</InputLabel>
-          <TextField
-            size='large'
-            clearable
-            value={name}
-            onChange={setName}
-            hasError={nameError != null}
-            helperText={nameError}
-          />
-        </FormField>
-        <FormField>
-          <InputLabel>{t.timingInputLabel}</InputLabel>
-          <RadioField
-            value={type}
-            onChange={setType}
-            name='type'
-            options={[
-              {
-                label: t.typeOngoingLabel,
-                value: 'ongoing',
-              },
-              {
-                label: t.typeTimeBoundLabel,
-                value: 'time_bound',
-              },
-            ]}
-          />
-          <>
-            {type === 'time_bound' && (
-              <AccessCodeFormTimes
-                startDate={startDate}
-                endDate={endDate}
-                onEdit={() => {
-                  setDatePickerVisible(true)
-                }}
-              />
-            )}
-          </>
-        </FormField>
-        <div className='seam-actions'>
-          <Button onClick={onBack}>{t.cancel}</Button>
-          <Button variant='solid' disabled={!isFormValid} onMouseDown={submit}>
-            {t.save}
-          </Button>
-        </div>
-      </div>
-    </>
-  )
-}
-
-function useSubmit(params: {
-  name: string
-  type: AccessCode['type']
-  device: NonNullable<UseDeviceData>
-  startDate: string
-  endDate: string
-  timezone: string
-  onSuccess?: () => void
-}): {
-  submit: () => void
-  isLoading: boolean
-} {
-  const { name, type, device, startDate, endDate, timezone, onSuccess } = params
-
-  const { mutate, isLoading } = useCreateAccessCode()
-  const submit = (): void => {
-    if (name === '') {
-      return
-    }
-
-    if (isLoading) {
-      return
-    }
-
-    if (type === 'time_bound') {
-      mutate(
-        {
-          name,
-          device_id: device.device_id,
-          starts_at: createIsoDate(startDate, timezone),
-          ends_at: createIsoDate(endDate, timezone),
-        },
-        {
-          onSuccess,
-        }
-      )
-
-      return
-    }
-
-    mutate(
-      {
-        name,
-        device_id: device.device_id,
-      },
-      {
-        onSuccess,
-      }
     )
   }
 
-  return { submit, isLoading }
-}
-
-const t = {
-  addNewAccessCode: 'Add new access code',
-  overCharacterLimitError: '60 characters max',
-  nameInputLabel: 'Name the new code',
-  cancel: 'Cancel',
-  save: 'Save',
-  timingInputLabel: 'Timing',
-  typeOngoingLabel: 'Ongoing',
-  typeTimeBoundLabel: 'Start/end times',
+  throw new Error(
```

Comment:
Yea, I've moved things around in https://github.com/seamapi/react/pull/312/commits/140a430f4424af96a7663ef3af2bc6edf841c9cb. Instead of having the generic `<AccessCodeForm/>` be at the top-level, users should use the more specific components:

- `<CreateAccessCodeForm/>`
- `<EditAccessCodeForm/>`

If they want to handle the submission themselves, they can of course use the more low-level `<AccessCodeForm/>`.

The above structure has made all the complexity of types, and cross-cutting hooks disappear. It does deviate a bit from the spec though.

Reply:
Nice. I am thinking we should just have `CreateAccessCodeForm` and `EditAccessCodeForm` and not export `AccessCodeForm`. I know that was not the original thought, but it's a cleaner api IMO. Same reason we have `/access_code/update` and `/access_code/create`.

Reply:
We could do. Should I rename / add the stories for this?

Reply:
We should confirm with @dawnho first. I strongly recommend this approach.

Reply:
If we go this route, I guess we should move `AccessCodeForm` to the ui folder and have `CreateAccessCodeForm` and `EditAccessCodeForm` as their own folders / elements just like the other main components.

Reply:
@mikewuu All good to go ahead with the plan.

Reply:
Refactored in https://github.com/seamapi/react/pull/312/commits/bebc0a36ef6a1b5fc6710c51d4c5e30db704d54d